### PR TITLE
Speeeeed cleanup

### PR DESCRIPTION
**Pull request by o-**
cleanup of #329 

deals with the open review comments.

additionally this commit finally moves the interpreter into the rir namespace...

### CODE REVIEW

**Comment by charig** [comment]
Do not know if this would add any overhead but definitely it simplifies a lot code browsing

**Review by Jakobeha — COMMENTED** [review]
The `interp_` prefixes seem redundant, maybe just have`context`, `data`, and `builtins`.

The naming of `interp_context` and `interp_data` is a little confusing. `interp_data` mostly defines `CallContext` and some constants for arguments. Maybe rename it to `interp_call`, `interp_fun`, or `call_context`, and move some of the constants into other files.

File: rir/src/interpreter/runtime.h [hunk 1]
```diff
@@ -26,6 +26,7 @@ void initializeRuntime();
   shared - it is not that we add stuff to them often.
  */
 Context* globalContext();
-rir::Configurations* pirConfigurations();
+Configurations* pirConfigurations();
+}
```
Comment (mhyee) [inline rir/src/interpreter/runtime.h]: `// namespace rir`

File: rir/src/interpreter/runtime.cpp [hunk 2]
```diff
@@ -41,4 +43,5 @@ void initializeRuntime() {
 }
 
 Context* globalContext() { return globalContext_; }
-rir::Configurations* pirConfigurations() { return configurations; }
+Configurations* pirConfigurations() { return configurations; }
+}
```
Comment (mhyee) [inline rir/src/interpreter/runtime.cpp:47]: `// namespace rir`

File: rir/src/interpreter/interp_context.h [hunk 3]
```diff
@@ -247,5 +249,6 @@ RIR_INLINE void cp_pool_set(Context* c, unsigned index, SEXP e) {
     SLOWASSERT(c->cp.capacity > index);
     SET_VECTOR_ELT(c->cp.list, index, e);
 }
+}
```
Comment (mhyee) [inline rir/src/interpreter/interp_context.h]: `// namespace rir`

File: rir/src/interpreter/interp_context.cpp [hunk 4]
```diff
@@ -69,3 +69,4 @@ Context* context_create() {
 }
 
 extern Context* globalContext_;
+}
```
Comment (mhyee) [inline rir/src/interpreter/interp_context.cpp]: `// namespace rir`

File: rir/src/interpreter/interp.h [hunk 5]
```diff
@@ -26,8 +23,6 @@ SEXP rirApplyClosure(SEXP, SEXP, SEXP, SEXP);
 
 SEXP argsLazyCreation(void* rirDataWrapper);
 
-#ifdef __cplusplus
 }
```
Comment (mhyee) [inline rir/src/interpreter/interp.h]: `// namespace rir`

File: rir/src/interpreter/interp.cpp [hunk 6]
```diff
@@ -3526,3 +3513,4 @@ SEXP rirEval_f(SEXP what, SEXP env) {
 
     assert(false && "Expected a code object or a dispatch table");
 }
+}
```
Comment (mhyee) [inline rir/src/interpreter/interp.cpp:3151]: `// namespace rir`

**Review by mhyee — COMMENTED** [review]
Agree that `interp_builtins` could just be renamed to `builtins`.

I also never really liked `interp_data` and `interp_context`, but I'm not sure what a better name is.

File: rir/src/interpreter/interp.cpp (hunk 6 above)
Comment (o-) [reply rir/src/interpreter/interp.cpp:3151]: bah. I wonder why sometimes clang-format does that for me and sometimes not :(

**Pull request merged**